internal/receiver/discoveryreceiver/endpoint_tracker_test.go

```diff
@@ -16,6 +16,7 @@ package discoveryreceiver
 
 import (
 	"path"
+	"sync"
```

Comment:
With the addition of more goroutine logic, we should add `goleak` to this package.

Reply:
added